src/udev/udev-builtin-net_id.c

```diff
@@ -160,6 +166,49 @@ static struct udev_device *skip_virtio(struct udev_device *dev) {
         return parent;
 }
 
+static int get_virtfn_info(struct udev_device *dev, struct netnames *names, struct virtfn_info *vf_info) {
+        struct udev *udev = udev_device_get_udev(names->pcidev);
+        const char *physfn_link_file, *virtfn_link_file;
+        _cleanup_free_ char *physfn_pci_syspath = NULL;
+        _cleanup_free_ char *virtfn_pci_syspath = NULL;
+        struct dirent *dent;
+        DIR *dir = NULL;
+
+        /* Check if this is a virtual function. */
+        physfn_link_file = strjoina(udev_device_get_syspath(names->pcidev), "/physfn");
+        if (readlink_and_canonicalize(physfn_link_file, NULL, &physfn_pci_syspath))
```

Comment:
Now readlink_and_canonicalize() is dropped. Please use chase_symlinks() instead.